dlc-manager/src/sub_channel_manager.rs

```diff
@@ -2899,9 +2917,11 @@ where
             let mut sub_channel = match self
                 .dlc_channel_manager
                 .get_store()
-                .get_sub_channel(channel_info.channel_id)?
+                .get_sub_channel(SubChannelId::from_bytes(channel_info.channel_id))?
             {
                 None => {
+                    // TODO: An error might be too much, especially since we will get
+                    // `DlcChannelId`s here even though they are supposed to be handled elsewhere!
```

Comment:
I don't get what you're saying here.

Reply:
My understanding is that `confirmed_txs` will return confirmed transactions for both subchannels and DLC channels because they're registered based on:

```rust
pub(crate) struct ChannelInfo {
    /// The identifier for _either_ a Lightning channel or a DLC channel.
    pub channel_id: [u8; 32],
    pub tx_type: TxType,
}
```

So if a transaction registered for a DLC channel (based on a `DlcChannelId`) is confirmed and has not been cleaned up yet (via `cleanup_channel`), it will pop up here and we will not be able to find a subchannel in the store for it, which will lead to us printing this error unnecessarily.

Alternatively we could introduce a `ChannelKind` enum in `chain_monitor.rs` so that the caller can, for example, filter out the confirmed transactions for irrelevant channel kinds. That way we can avoid misleading log messages.

Reply:
Ok got it thanks! I feel a proper fix would be to have each manager having its own channel monitor, but being updated using the same data (so for example you'd only process blocks once).

Reply:
I totally agree with you comment, but I guess such a fix doesn't belong in this PR.

I will just remove the TODO and create an issue that we can address later.

Reply:
Issue here: https://github.com/p2pderivatives/rust-dlc/issues/193.